app/models/concerns/due_date_queries.rb

```diff
@@ -1,291 +1,45 @@
 # frozen_string_literal: true
 
 module DueDateQueries
-  extend ActiveSupport::Concern
-
-  included do
-    # Scopes for common deadline queries
-    scope :upcoming, -> { where('due_at > ?', Time.current).order(:due_at) }
-    scope :overdue, -> { where('due_at < ?', Time.current).order(:due_at) }
-    scope :today, -> { where(due_at: Time.current.beginning_of_day..Time.current.end_of_day) }
-    scope :this_week, -> { where(due_at: Time.current.beginning_of_week..Time.current.end_of_week) }
-    scope :for_deadline_type, ->(type_name) { joins(:deadline_type).where(deadline_types: { name: type_name }) }
-    scope :for_round, ->(round_num) { where(round: round_num) }
-    scope :active_deadlines, -> { where('due_at > ?', Time.current) }
-    scope :by_deadline_type, -> { joins(:deadline_type).order('deadline_types.name') }
-  end
-
-  class_methods do
-    # Find next upcoming deadline for any parent
-    def next_deadline
-      upcoming.first
-    end
-
-    # Find deadlines by type name
-    def of_type(deadline_type_name)
-      joins(:deadline_type).where(deadline_types: { name: deadline_type_name })
-    end
-
-    # Get deadline statistics
-    def deadline_stats
-      {
-        total: count,
-        upcoming: upcoming.count,
-        overdue: overdue.count,
-        today: today.count,
-        this_week: this_week.count
-      }
-    end
-
-    # Find deadlines within a date range
-    def between_dates(start_date, end_date)
-      where(due_at: start_date..end_date)
-    end
-
-    # Find deadlines for specific actions
-    def for_submission
-      of_type('submission')
-    end
-
-    def for_review
-      of_type('review')
-    end
-
-    def for_quiz
-      of_type('quiz')
-    end
-
-    def for_teammate_review
-      of_type('teammate_review')
-    end
-
-    def for_metareview
-      of_type('metareview')
-    end
-
-    # Find deadlines that allow specific actions
-    def allowing_submission
-      where(submission_allowed_id: [2, 3]) # Late and OK
-    end
-
-    def allowing_review
-      where(review_allowed_id: [2, 3]) # Late and OK
-    end
-
-    def allowing_quiz
-      where(quiz_allowed_id: [2, 3]) # Late and OK
-    end
-
-    # Get deadlines grouped by type
-    def grouped_by_type
-      joins(:deadline_type)
-        .group('deadline_types.name')
-        .order('deadline_types.name')
-    end
-  end
-
-  # Instance methods for parent objects (Assignment, SignUpTopic)
-  # These methods should be included in Assignment and SignUpTopic models
-
   # Get next due date for this parent
```

Comment:
I assume that "topic_id" here means the primary key of the topic in the topics table.
It is not guaranteed that all topics will have user-assigned numbers like "E2566".